rdrf/rdrf/views/form_view.py

```diff
@@ -530,7 +533,7 @@ def post(self, request, registry_code, form_id, patient_id, context_id=None):
                                                      patient_id)
         if request.user.is_superuser:
             pass
-        elif request.user.is_working_group_staff or request.user.has_perm("rdrf.form_%s_is_readonly" % form_id):
+        elif request.user.is_working_group_staff or request.user.is_readonly(form_obj):
```

Comment:
Seeing as `allowed_groups` works basically the same way, maybe move the readonly check into `can_view`, or move the `allowed_groups` check into the same place as the readonly check

Reply:
As discussed offline, we still need the two separate methods as they are used for different purposes.